### ISSUE TITLE
New-AzFunctionApp always returns "Function name... is not available'

### ISSUE THREAD

**Issue by keith-knight** [Azure/azure-powershell#16448]
### Description

I have been trying to create an Azure Function App (version 4) using New-AzFunctionApp for example:

`New-AzFunctionApp -Name MyFunctionAppNameHere -ResourceGroupName MyResourceGroupNameHere -Location australiaeast -StorageAccount MyStorageAccountNameHere -Runtime "dotnet-isolated"-FunctionsVersion 4`

Every time it returns the following error:

```
InvalidOperation: C:\Users\keith\Documents\PowerShell\Modules\Az.Functions\3.1.1\custom\HelperFunctions.ps1:807
Line |
 807 |      throw $errorRecord
     |      ~~~~~~~~~~~~~~~~~~
     | Function name 'MyFunctionAppNameHere ' is not available.  Please try a different name.
```

I have been able to successfully create a function app using the Azure CLI with the same parameters. I have also logged into the Azure portal and typed the function name into the create function screen to verify that the name is not in use.

I have tried a number of different variations on the above command - for example using "Australia East" rather than "australiaeast" as the location, trying different permutations of runtime and function version and the error is always the same as above... and obviously trying a number of different names, all of which were not actually in use (verified using the portal).

### Issue script & Debug output

```PowerShell
DEBUG: 3:16:08 PM - GetAzureRMContextCommand end processing.
DEBUG: [CmdletBeginProcessing]: Starting command
DEBUG: CmdletBeginProcessing:
DEBUG: CmdletProcessRecordStart:
DEBUG: [CmdletProcessRecordAsyncStart]: Created new QosEvent for command 'Test-AzNameAvailability_CheckExpanded'
DEBUG: CmdletProcessRecordAsyncStart:
DEBUG: CmdletGetPipeline:
DEBUG: CmdletBeforeAPICall:
DEBUG: URLCreated: /subscriptions/mysub-id-here/providers/Microsoft.Web/checknameavailability?api-version=2019-08-01
DEBUG: RequestCreated: /subscriptions/mysub-id-here/providers/Microsoft.Web/checknameavailability?api-version=2019-08-01
DEBUG: HeaderParametersAdded:
DEBUG: BodyContentSet:
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
POST

Absolute Uri:
https://management.azure.com/subscriptions/mysub-id-here/providers/Microsoft.Web/checknameavailability?api-version=2019-08-01

Headers:
x-ms-unique-id                : 16
x-ms-client-request-id        : b7beeed8-9a93-4c8b-b86a-f221f2efa6e0
CommandName                   : Az.Functions.internal\Test-AzNameAvailability
FullCommandName               : Test-AzNameAvailability_CheckExpanded
ParameterSetName              : __AllParameterSets
User-Agent                    : AzurePowershell/v6.6.0,PSVersion/v7.2.0

Body:
{
  "name": "MyFunctionAppNameHere",
  "type": "Site"
}


DEBUG: BeforeCall:
DEBUG: Finally:
DEBUG: [CmdletProcessRecordAsyncEnd]: Finish HTTP process
DEBUG: CmdletProcessRecordAsyncEnd:
DEBUG: CmdletProcessRecordEnd:
DEBUG: AzureQoSEvent: Module: Az.Functions:3.1.1.0; CommandName: Test-AzNameAvailability_CheckExpanded; PSVersion: 7.2.0; IsSuccess: True; Duration: 00:00:00
DEBUG: Finish sending metric.
DEBUG: CmdletEndProcessing:
```

### Environment data

```PowerShell
Name                           Value
----                           -----
PSVersion                      7.2.0
PSEdition                      Core
GitCommitId                    7.2.0
OS                             Microsoft Windows 10.0.19043
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Module versions

```PowerShell
ModuleType Version    PreRelease Name                                ExportedCommands
---------- -------    ---------- ----                                ----------------
Script     2.6.2                 Az.Accounts                         {Add-AzEnvironment, Clear-AzContext, Clear-AzDefa…
Script     3.1.1                 Az.Functions                        {Get-AzFunctionApp, Get-AzFunctionAppAvailableLoc…
Script     4.4.1                 Az.Resources                        {Add-AzADGroupMember, Export-AzResourceGroup, Exp…
Script     3.12.0                Az.Storage                          {Add-AzRmStorageContainerLegalHold, Add-AzStorage…
```

### Error output

```PowerShell
HistoryId: -1

Message        : Function name 'MyFunctionAppNameHere' is not available.  Please try a different name.
StackTrace     :
Exception      : System.InvalidOperationException
InvocationInfo : {}
Line           :     throw $errorRecord

Position       : At C:\Users\keith\Documents\PowerShell\Modules\Az.Functions\3.1.1\custom\HelperFunctions.ps1:807
                 char:5
                 +     throw $errorRecord
                 +     ~~~~~~~~~~~~~~~~~~
```

**Comment by dingmeng-xue** [comment]
Thank you for your reporting. Tagging and routing to the team member best able to assist.

**Comment by SaurabhSharma-MSFT** [comment]
@keith-knight Have you tried executing `New-AzFunctionApp` multiple times with the same function name or is this the first time you run with the same name and got this error ?

**Comment by keith-knight** [comment]
Hi @SaurabhSharma-MSFT, I'm unsure what changed but after coming back to it after the weekend it is now working fine. I needed to change a few of the other parameter values (e.g. the runtime) but have not since run into the issue with the function name being in use.

`New-AzFunctionApp -Name $ServicesFunctionAppName -ResourceGroupName $ResourceGroupName -Location $GeoLocation -StorageAccount $StorageAccountName -Runtime "DotNet" -FunctionsVersion 4 -RuntimeVersion 6`

When it was failing I tried a number of different function names (at least 5 of them), each time using the Azure Portal to verify that the name was not in use. Also even with the invalid runtime value of "dotnet-isolated" it always came back with the "function name not available" not the error "Runtime 'dotnet-isolated' is not supported." that it returns now with the invalid runtime.

Thanks for your help.
